### ISSUE TITLE
Stimulus Binds Twice on New Page after Redirect to GET Request

### ISSUE THREAD

**Issue by rience** [hotwired/stimulus#629]
Hi,

I've observed this strange behaviour and I couldn't find information about that in documentation. Stimulus does connect, disconnect and connect again page user is redirected to - depends on what was method used in previous request.

Test scenario

1. Have controller's action that responds with redirect to new page (the one with Stimulus controller).
2. Bind this action to GET request - observe in console that Stimulus will connect, disconnect and connect again.
3. Now change this action to POST - observe that Stimulus will only connect once.

This is in Rails 7 application with Turbo framework. However, I've opted-out of caching there.

You can find application that demonstrates this behaviour here: https://github.com/rience/stimulus-double-bind

Thank you in advance for help.

**Comment by marcoroth** [comment]
Hey @rience, thanks for reporting this issue and for providing a repro!

I just cloned your repo and can confirm that I can reproduce it. To me this seems like a Turbo issue 🤔 But nonetheless, I wanted to report what I see.

If you additionally log the controller element of the Stimulus controller you'll notice that the first connect/disconnect references another element which is not present anymore after the third log appears. 

```js
  connect() {
    console.log("hello_controller: connect()", this.element);
  }
```

My current best guess is that Turbo is doing something weird while trying to append the new fetched GET page via Turbo. Seems like this is where this double logs originates from.

**Comment by rience** [comment]
Hi @marcoroth,

Yes - I thought about opening this ticket in Turbo repo - but only after I opened it here. So maybe I'll close this one and I'll open new one in Turbo?

**Comment by marcoroth** [comment]
Thanks @rience! I'm wondering if it's related to how Turbo processes snapshots. I'll investigate.

**Comment by marcoroth** [comment]
Closing as duplicate of https://github.com/hotwired/turbo/issues/794

This one might be fixed by https://github.com/hotwired/turbo/pull/804
